### PR TITLE
Bugfix/mw reduced repl deadlocks

### PR DESCRIPTION
**Pull request by lordnull**
Fixes to the mutate_get for reduced repl to remove several instances that could deadlock a vnode. The problems are apparent when running the repl_reduced:read_repair_interaction riak test as found on https://github.com/basho/riak_test/pull/392 .

### CODE REVIEW

File: src/riak_repl_ring_handler.erl [hunk 1]
```diff
@@ -200,4 +202,13 @@ rt_update_events(Ring) ->
...
     riak_repl2_pg:ensure_pg(riak_repl_ring:pg_enabled(Ring)).
 
+update_mutator(_Ring) ->
+    case riak_core_capability:get({riak_kv, mutators}, false) of
+        false ->
+            lager:debug("unregister repl reduced"),
+            riak_kv_mutator:unregister(riak_repl_reduced);
+        true ->
+            lager:debug("registering repl reduced"),
+            riak_kv_mutator:register(riak_repl_reduced, 5)
```
Comment (cmeiklejohn) [inline src/riak_repl_ring_handler.erl]: Why 5?  Should this be defined in a header file somewhere?

File: src/riak_repl_ring_handler.erl [hunk 2]
```diff
@@ -200,4 +202,13 @@ rt_update_events(Ring) ->
 pg_update_events(Ring) ->
     riak_repl2_pg:ensure_pg(riak_repl_ring:pg_enabled(Ring)).
 
+update_mutator(_Ring) ->
```
Comment (cmeiklejohn) [inline src/riak_repl_ring_handler.erl]: It definitely seems like this function could be named better.  In addition, a doc string detailing why it's doing what it is would also be nice.

**Comment by cmeiklejohn** [comment]
I'm still getting failures when running the test above using this branch.

File: src/riak_repl_reduced.erl [hunk 3]
```diff
@@ -200,6 +212,26 @@ mutate_put(InMeta, InVal, RevealedMeta, RObj, BucketProps, NumberReals) ->
             maybe_reduce(InMeta, InVal, RevealedMeta, RObj, RealsList, NumberReals)
     end.
 
+% the symbolic_clustername functions in riak_core_connection always return
```
Comment (cmeiklejohn) [inline src/riak_repl_reduced.erl:214]: Add `@doc`.

File: src/riak_repl_reduced.erl [hunk 4]
```diff
@@ -200,6 +212,26 @@ mutate_put(InMeta, InVal, RevealedMeta, RObj, BucketProps, NumberReals) ->
...
+% a string. However, if the cluster is not named, that string is useless
+% for proxy get. So, rather than do a proxy get that can never succeede
```
Comment (cmeiklejohn) [inline src/riak_repl_reduced.erl]: Typo in succeed.

File: src/riak_repl_reduced.erl [hunk 5]
```diff
@@ -93,15 +106,13 @@ local_ring_get(InObject, BKey, Partition) ->
     riak_core_vnode_master:command(Preflist, Req, {raw, ReqId, self()}, riak_kv_vnode_master),
     receive
         {ReqId, {r, {ok, RObj}, _, ReqId}} ->
+            %lager:debug("successful get!"),
```
Comment (cmeiklejohn) [inline src/riak_repl_reduced.erl]: Commented code.

File: src/riak_repl_reduced.erl (hunk 3 above)
Comment (lordnull) [inline src/riak_repl_reduced.erl:214]: Even if it's not an exported function?

**Comment by lordnull** [comment]
No longer relevant due to mutators being removed.

**Pull request closed without merging**